### PR TITLE
[RFC-774] Deprecate implicit record loading in Ember Route

### PR DESCRIPTION
**Pull request by wagenet**
[RFC-774](https://github.com/emberjs/rfcs/pull/774)

Deprecation App: https://github.com/ember-learn/deprecation-app/pull/1334

### CODE REVIEW

File: packages/@ember/routing/route.ts [hunk 1]
```diff
@@ -1400,12 +1401,20 @@ class Route<T = unknown>
     if a promise returned from `model` fails, the error will be
     handled by the `error` hook on `Route`.
 
+    Note that the legacy behavior of automatically defining a model
+    hook when a dynamic segment ending in `_id` is present is deprecated.
+    You should explicitly define a model hook whenever any segments are
+    present.
```
Comment (wagenet) [inline packages/@ember/routing/route.ts:1407]: Add deprecation id or 6.0 mention here so we make sure to remove this in the future.

File: packages/@ember/routing/route.ts [hunk 2]
```diff
@@ -1816,10 +1853,23 @@ class Route<T = unknown>
...
   @computed
   protected get store() {
+    deprecate(
+      `The default store behavior for routes is deprecated. Please define an ` +
+        `explicit store for ${this.fullRouteName}, such as with \`@service store\`.`,
+      this._disableStoreDeprecation,
+      {
+        id: 'deprecate-defaul-route-store',
+        for: 'ember-source',
+        since: { available: '4.12.0' },
```
Comment (wagenet) [inline packages/@ember/routing/route.ts:1868]:
```suggestion
        since: { available: '5.0' },
```

File: packages/@ember/routing/route.ts [hunk 3]
```diff
@@ -1479,8 +1488,25 @@ class Route<T = unknown>
...
-    return (get(this, 'store') as any).find(...args);
+  findModel(type: string, value: unknown) {
+    deprecate(
+      `The implicit model loading behavior for routes is deprecated. ` +
+        `Please define an explicit model hook for ${this.fullRouteName}.`,
+      false,
+      {
+        id: 'deprecate-implicit-route-model',
+        for: 'ember-source',
+        since: { available: '4.12.0' },
```
Comment (wagenet) [inline packages/@ember/routing/route.ts:1499]:
```suggestion
        since: { available: '5.0.0' },
```

File: packages/@ember/routing/route.ts [hunk 4]
```diff
@@ -1233,14 +1243,24 @@ class Route<Model = unknown> extends EmberObject.extend(ActionHandler, Evented)
...
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let store = ('store' in this ? this.store : get(this, '_store')) as any;
-    return store.find(...args);
+  findModel(type: string, value: unknown) {
+    deprecate(
+      `The implicit model loading behavior for routes is deprecated. ` +
+        `Please define an explicit model hook for ${this.fullRouteName}.`,
+      false,
+      {
+        id: 'deprecate-implicit-route-model',
```
Comment (kategengler) [inline packages/@ember/routing/route.ts:1252]: I see there is a deprecation guide -- I think that should be here in the `url` property?

Comment (bertdeblock) [reply packages/@ember/routing/route.ts:1252]: @kategengler => https://github.com/emberjs/ember.js/pull/20524.

**Pull request merged**